### PR TITLE
Add cord 19 dataset

### PR DESCRIPTION
**Pull request by ggdupont**
Initial version only reading the metadata in CSV.

### Checklist:
- [x] Create the dataset script /datasets/my_dataset/my_dataset.py using the template
- [x] Fill the _DESCRIPTION and _CITATION variables
- [x] Implement _infos(), _split_generators() and _generate_examples()
- [x] Make sure that the BUILDER_CONFIGS class attribute is filled with the different configurations of the dataset and that the BUILDER_CONFIG_CLASS is specified if there is a custom config class.
- [x] Generate the metadata file dataset_infos.json for all configurations
- [x] Generate the dummy data dummy_data.zip files to have the dataset script tested and that they don't weigh too much (<50KB)
- [x] Add the dataset card README.md using the template and at least fill the tags
- [x] Both tests for the real data and the dummy data pass.

### Extras:
- [x] add more metadata
- [x] add full text
- [x] add pre-computed document embedding

### CODE REVIEW

**Comment by ggdupont** [comment]
Cleaned-up version of previous PR: https://github.com/huggingface/datasets/pull/1129

**Comment by ggdupont** [comment]
@lhoestq FYI

File: datasets/cord19/README.md [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+### Languages
+
+The dataset is in english (en).
+
+## Dataset Structure
+
+### Data Instances
+
+The following code block present an overview of a sample in json-like syntax (abbreviated since some fields are very long):
+```json
```
Comment (lhoestq) [inline datasets/cord19/README.md:88]:
````suggestion
```
````

File: datasets/cord19/README.md [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+- `s2_id`:  A `str`-valued field containing the Semantic Scholar ID for this paper.  Can be used with the Semantic Scholar API (e.g. `s2_id=9445722` corresponds to `http://api.semanticscholar.org/corpusid:9445722`)
+
+Extra fields based on selected configuration during loading:
+
+- `fulltext`: A `str`-valued field containing the concatenation of all text sections from json (itself extracted from pdf)
+- `doc_embeddings`: A `sequence` of float-valued elements containing document embeddings as a vector of floats (parsed from string of values separated by ','). Details on the system used to extract the embeddings are available in: [SPECTER: Document-level Representation Learning using Citation-informed Transformers](https://arxiv.org/abs/2004.07180). TL;DR: it's relying on a BERT model pre-trained on document-level relatedness using the citation graph. The system can be queried through REST (see [public API documentation](https://github.com/allenai/paper-embedding-public-apis)).
+
+### Data Splits
+
+No annotation provided in this dataset so all instances are provided in training split.
```
Comment (lhoestq) [inline datasets/cord19/README.md:145]:
```suggestion
No annotation provided in this dataset so all instances are provided in training split.

The sizes of each configurations are:

|          | Tain   |
| -----    | ------ |
| metadata | 368618  |
| fulltest | 368618  |
| embeddings | 368618  |

```

File: datasets/cord19/cord19.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+            embeddings_dir_path = os.path.join(data_dir, CORD19_DATASET_DATE + "/cord_19_embeddings")
+            if not os.path.isdir(embeddings_dir_path):
+                embeddings_dir_path = dl_manager.extract(
+                    os.path.join(data_dir, CORD19_DATASET_DATE + "/cord_19_embeddings.tar.gz")
+                )
+            files["embeddings"] = os.path.join(
+                embeddings_dir_path, "cord_19_embeddings_" + CORD19_DATASET_DATE + ".csv"
+            )
+
+        print("0000000000 {}".format(files))
```
Comment (lhoestq) [inline datasets/cord19/cord19.py:143]:
```suggestion
```
Can you remove the print ?

File: datasets/cord19/cord19.py [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+                    doc_id = data[0]
+
+                    doc_fields["doc_embeddings"] = []
+
+                    if doc_id == doc_fields["cord_uid"]:
+                        doc_fields["doc_embeddings"] = [float(v) for v in data[1:-1]]
+
+                yield i, doc_fields
+
+            if "embeddings" and fh is not None:
```
Comment (lhoestq) [inline datasets/cord19/cord19.py:216]:
```suggestion
            if "embeddings" in self.config.name and fh is not None:
```

**Review by lhoestq — APPROVED** [review]
Nice thank you !

**Comment by lhoestq** [comment]
Before merging I might tweak a little bit the dummy data to avoid having to check if the `document_parses` and `embeddings` directories exist or not. I'll do that later today

File: datasets/cord19/README.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+@article{Wang2020CORD19TC,
+  title={CORD-19: The Covid-19 Open Research Dataset},
+  author={Lucy Lu Wang and Kyle Lo and Yoganand Chandrasekhar and Russell Reas and Jiangjiang Yang and Darrin Eide and
+  K. Funk and Rodney Michael Kinney and Ziyang Liu and W. Merrill and P. Mooney and D. Murdick and Devvret Rishi and
+  Jerry Sheehan and Zhihong Shen and B. Stilson and A. Wade and K. Wang and Christopher Wilhelm and Boya Xie and
+  D. Raymond and Daniel S. Weld and Oren Etzioni and Sebastian Kohlmeier},
+  journal={ArXiv},
+  year={2020}
+}
+```
```
Comment (lhoestq) [inline datasets/cord19/README.md:226]:
````suggestion
```

### Contributions

Thanks to [@ggdupont](https://github.com/ggdupont) for adding this dataset.
````
we have a new section now :)

File: datasets/cord19/README.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+  - [Annotations](#annotations)
+  - [Personal and Sensitive Information](#personal-and-sensitive-information)
+- [Considerations for Using the Data](#considerations-for-using-the-data)
+  - [Social Impact of Dataset](#social-impact-of-dataset)
+  - [Discussion of Biases](#discussion-of-biases)
+  - [Other Known Limitations](#other-known-limitations)
+- [Additional Information](#additional-information)
+  - [Dataset Curators](#dataset-curators)
+  - [Licensing Information](#licensing-information)
+  - [Citation Information](#citation-information)
```
Comment (lhoestq) [inline datasets/cord19/README.md:61]:
```suggestion
  - [Citation Information](#citation-information)
  - [Contributions](#contributions)
```

**Review by lhoestq — COMMENTED** [review]
Adding the new section in the readme

**Comment by lhoestq** [comment]
Looks all good now ! Thanks a lot @ggdupont :)
Merging

**Pull request merged**